File: app/src/main/java/com/readrops/app/database/Converters.java
package com.readrops.app.database;

import androidx.room.TypeConverter;

import com.readrops.app.database.entities.Account;

import org.joda.time.LocalDateTime;

public class Converters {

    @TypeConverter
    public LocalDateTime fromTimeStamp(Long value) {
        return new LocalDateTime(value);
    }

    @TypeConverter
    public long fromLocalDateTime(LocalDateTime localDateTime) {
        return localDateTime.toDateTime().getMillis();
    }

    @TypeConverter
    public Account.AccountType fromAccountTypeCode(int ordinal) {
        return Account.AccountType.values()[ordinal];
    }

    @TypeConverter
    public int getAccountTypeCode(Account.AccountType accountType) {
        return accountType.ordinal();
    }

}

File: db/src/main/java/com/readrops/db/Converters.kt
package com.readrops.db

import androidx.room.TypeConverter
import com.readrops.db.entities.account.AccountType
import org.joda.time.LocalDateTime

class Converters {

    @TypeConverter
    fun fromTimeStamp(value: Long): LocalDateTime {
        return LocalDateTime(value)
    }

    @TypeConverter
    fun fromLocalDateTime(localDateTime: LocalDateTime): Long {
        return localDateTime.toDateTime().millis
    }

    @TypeConverter
    fun fromAccountTypeCode(ordinal: Int): AccountType {
        return AccountType.values()[ordinal]
    }

    @TypeConverter
    fun getAccountTypeCode(accountType: AccountType): Int {
        return accountType.ordinal
    }
}